TheMET/TheMET/Network/MetAPI.swift

```diff
@@ -164,22 +164,22 @@ class MetAPI{
             }
     }
     
-    func search(parameters: [SearchParameter], completion: @escaping (SearchResponse?) -> Void) {
+    func search(parameters: [SearchParameter], completion: @escaping (Result<SearchResponse, MetAPIError>) -> Void) {
         self.metAPICache.search(parameters: parameters) { [weak self] searchResponce in
             guard let self = self else {
-                completion(nil)
+                completion(.failure(.metAPIdoesNotAnswer))
                 return
             }
             if let searchResponce = searchResponce {
-                completion(searchResponce)
+                completion(.success(searchResponce))
             } else {
                 self.executeSearch(parameters: parameters) {[weak self] realResponse in
                     guard let realResponse = realResponse else {
-                        completion(nil)
+                        completion(.failure(.noDataInResponce))
```

Comment:
You still rely on `nil`/`non-nil` state of the `realResponse`. Please introduce `Result` to `func executeSearch` as well